File: 800/CF2245A_Who_Watches_the_Watchpig.java

import java.util.*;

public class CF2245A_Who_Watches_the_Watchpig {
    static final int INF = 1_000_000;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();

        while (T-- > 0) {
            int n = sc.nextInt();
            int k = sc.nextInt();
            String s = sc.next();

            int answer = INF;

            for (int m = 0; m <= n; m++) {

                // impossible if not enough R or L exist
                if (m < k || (n - m) < k)
                    continue;

                int[][] dp = new int[n + 1][m + 2];

                for (int i = 0; i <= n; i++)
                    Arrays.fill(dp[i], INF);

                dp[0][0] = 0;

                for (int i = 0; i < n; i++) {
                    for (int r = 0; r <= Math.min(i, m); r++) {

                        if (dp[i][r] == INF)
                            continue;

                        // Make current pig L
                        if (r >= k) {
                            int cost = dp[i][r] + (s.charAt(i) == 'R' ? 1 : 0);
                            dp[i + 1][r] = Math.min(dp[i + 1][r], cost);
                        }

                        // Make current pig R
                        if (r < m) {
                            int lBefore = i - r;
                            int lAfter = (n - m) - lBefore;

                            if (lAfter >= k) {
                                int cost = dp[i][r] + (s.charAt(i) == 'L' ? 1 : 0);
                                dp[i + 1][r + 1] = Math.min(dp[i + 1][r + 1], cost);
                            }
                        }
                    }
                }

                answer = Math.min(answer, dp[n][m]);
            }

            if (answer == INF)
                System.out.println(-1);
            else
                System.out.println(answer);
        }
    }
}

File: 800/CF2245A_Who_Watches_the_Watchpig.py
INF = 10 ** 9

t = int(input())

for _ in range(t):
    n, k = map(int, input().split())
    s = input().strip()

    ans = INF

    for m in range(n + 1):

        if m < k or (n - m) < k:
            continue

        dp = [[INF] * (m + 2) for _ in range(n + 1)]
        dp[0][0] = 0

        for i in range(n):
            for r in range(min(i, m) + 1):

                if dp[i][r] == INF:
                    continue

                # Make current pig L
                if r >= k:
                    cost = dp[i][r] + (s[i] == 'R')
                    dp[i + 1][r] = min(dp[i + 1][r], cost)

                # Make current pig R
                if r < m:
                    l_before = i - r
                    l_after = (n - m) - l_before

                    if l_after >= k:
                        cost = dp[i][r] + (s[i] == 'L')
                        dp[i + 1][r + 1] = min(dp[i + 1][r + 1], cost)

        ans = min(ans, dp[n][m])

    print(-1 if ans == INF else ans)